kyuubi-common/src/main/scala/org/apache/kyuubi/operation/log/OperationLog.scala

```diff
@@ -133,8 +133,10 @@ class OperationLog(path: Path) {
       var line: String = reader.readLine()
       while ((i < lastRows || maxRows <= 0) && line != null) {
         logs.add(line)
-        line = reader.readLine()
         i += 1
+        if (i < lastRows || maxRows <= 0) {
```

Comment:
thanks for catch this issue!

is `do { ... } while` pattern more suitable here?

Reply:
```
    val logs = new JArrayList[String]
    var rows = 0
    try {
      var line: String = null
      do {
        line = reader.readLine()
        if (line != null) { logs.add(line); rows += 1 }
      } while ((rows < lastRows || maxRows <= 0) && line != null)
      (logs, rows)
    } catch {
      ...
    }
```

Reply:
than you, this is better solution